dotnet/samples/Concepts/Agents/MixedChat_Streaming.cs

```diff
@@ -0,0 +1,157 @@
+                Name = ReviewerName,
+                Kernel = this.CreateKernelWithChatCompletion(),
+            };
+
+        OpenAIAssistantAgent agentWriter =
+            await OpenAIAssistantAgent.CreateAsync(
+                kernel: new(),
+                clientProvider: this.GetClientProvider(),
+                definition: new(this.Model)
+                {
+                    Instructions = CopyWriterInstructions,
+                    Name = CopyWriterName,
+                    Metadata = AssistantSampleMetadata,
+                });
+
+        // Create a chat for agent interaction.
+        AgentGroupChat chat =
+            new(agentWriter, agentReviewer)
+            {
+                ExecutionSettings =
+                    new()
+                    {
+                        // Here a TerminationStrategy subclass is used that will terminate when
+                        // an assistant message contains the term "approve".
+                        TerminationStrategy =
+                            new ApprovalTerminationStrategy()
+                            {
+                                // Only the art-director may approve.
+                                Agents = [agentReviewer],
+                                // Limit total number of turns
+                                MaximumIterations = 10,
+                            }
+                    }
+            };
+
+        // Invoke chat and display messages.
+        ChatMessageContent input = new(AuthorRole.User, "concept: maps made out of egg cartons.");
+        chat.AddChatMessage(input);
+        this.WriteAgentChatMessage(input);
+
+        string lastAgent = string.Empty;
+        await foreach (StreamingChatMessageContent response in chat.InvokeStreamingAsync())
+        {
+            if (string.IsNullOrEmpty(response.Content))
+            {
+                continue;
+            }
+
+            if (!lastAgent.Equals(response.AuthorName, StringComparison.Ordinal))
+            {
+                Console.WriteLine($"\n# {response.Role} - {response.AuthorName ?? "*"}:");
+                lastAgent = response.AuthorName ?? string.Empty;
+            }
+
+            Console.WriteLine($"\t > streamed: '{response.Content}'");
+        }
+
+        // Display the chat history.
+        Console.WriteLine("================================");
+        Console.WriteLine("CHAT HISTORY");
+        Console.WriteLine("================================");
+
+        ChatMessageContent[] history = await chat.GetChatMessagesAsync().Reverse().ToArrayAsync();
+
+        for (int index = 0; index < history.Length; index++)
+        {
+            this.WriteAgentChatMessage(history[index]);
+        }
+
+        Console.WriteLine($"\n[IS COMPLETED: {chat.IsComplete}]");
+    }
+
+    // Local function to invoke agent and display the conversation messages.
+    private async Task InvokeAgentAsync(ChatCompletionAgent agent, ChatHistory chat, string input)
```

Comment:
Looks like this local function contains the same code as above, should this function be used in place of that?

Reply:
This was copy/paste from another sample.  The method isn't even called.  Removed.